packages/vuetify/src/components/VDatePicker/mixins/date-picker-table.ts

```diff
@@ -0,0 +1,205 @@
+import '../../../stylus/components/_date-picker-table.styl'
+
+// Directives
+import Touch, { TouchWrapper } from '../../../directives/touch'
+
+// Mixins
+import Colorable from '../../../mixins/colorable'
+import Themeable from '../../../mixins/themeable'
+
+// Utils
+import isDateAllowed, { AllowedFunction } from '../util/isDateAllowed'
+import mixins from '../../../util/mixins'
+
+// Types
+import { VNodeChildren } from 'vue'
+import { PropValidator } from 'vue/types/options'
+import { NativeLocaleFormatter } from '../util/createNativeLocaleFormatter'
+
+type CalculateTableDateFunction = (v: number) => string
+
+type DateEventColorValue = string | string[]
+type DateEvents = string[] | ((date: string) => boolean | DateEventColorValue) | Record<string, DateEventColorValue>
+type DateEventColors = DateEventColorValue | Record<string, DateEventColorValue> | ((date: string) => DateEventColorValue)
+
+/* @vue/component */
+export default mixins(
+  Colorable,
+  Themeable
+).extend({
+  directives: { Touch },
+
+  props: {
+    allowedDates: {
+      type: Function
+    } as any as PropValidator<AllowedFunction | undefined>,
+    current: String,
+    disabled: Boolean,
+    format: {
+      type: Function
+    } as any as PropValidator<NativeLocaleFormatter | undefined>,
+    events: {
+      type: [Array, Function, Object],
+      default: () => null
+    } as any as PropValidator<DateEvents>,
+    eventColor: {
+      type: [Array, Function, Object, String],
+      default: () => 'warning'
+    } as any as PropValidator<DateEventColors>,
+    locale: {
+      type: String,
+      default: 'en-us'
+    },
+    min: String,
+    max: String,
+    readonly: Boolean,
+    scrollable: Boolean,
+    tableDate: {
+      type: String,
+      required: true
+    },
+    value: [String, Array]
+  },
+
+  data: () => ({
+    isReversing: false
+  }),
+
+  computed: {
+    computedTransition (): string {
+      return (this.isReversing === !this.$vuetify.rtl) ? 'tab-reverse-transition' : 'tab-transition'
+    },
+    displayedMonth (): number {
+      return Number(this.tableDate.split('-')[1]) - 1
+    },
+    displayedYear (): number {
+      return Number(this.tableDate.split('-')[0])
+    }
+  },
+
+  watch: {
+    tableDate (newVal: string, oldVal: string) {
+      this.isReversing = newVal < oldVal
+    }
+  },
+
+  methods: {
+    genButtonClasses (isAllowed: boolean, isFloating: boolean, isSelected: boolean, isCurrent: boolean) {
```

Comment:
Maybe we should specify return type for this functions?

Reply:
No need, the type is inferred